### DISCUSSION TITLE
help understanding ingest pipeline for ossec

### DISCUSSION THREAD

**Discussion by lmdeaton1** [Security-Onion-Solutions/securityonion#7873] [Unsupported Versions] [answered]
I've been studying the ingest pipeline definitions for the ossec pipeline ( wazuh )  as shown below:

{
  "description" : "ossec",
  "processors" : [
    { "json":           { "field": "message",                                   "target_field": "message2",             "ignore_failure": true  } },
    { "remove": { "field": [ "agent" ], "ignore_missing": true, "ignore_failure": true } },
    { "rename":         { "field": "message2.agent",                            "target_field": "agent",                "ignore_missing": true  } },
...

If I understand this, this is:
1) taking the field "message" and mapping into the "message2" field.  
2) it is removing the "agent" field from the json.  
3) it is renaming the "message2.agent" field into the "agent" field.  

Assuming I am understanding this correctly, can  someone explain to me why you would want to do this?  It seems as though the "agent" field would have the same values as it did originally.  It seems like a waste of processing.  What do you gain by doing this?  What am I missing?

**Comment by dougburks** [comment, marked as answer]
The original agent field that gets removed contains unnecessary Filebeat metadata. We remove that so we can replace it with more relevant Wazuh agent information.

**Reply by lmdeaton1** [reply]
But are you not copying the same agent field, including filebeat metadata, back when you rename "message2.agent" to the "agent" field?

**Reply by dougburks** [reply]
No, as I mentioned in my previous reply, it is **not** the same agent field and it does **not** include filebeat metadata. If you want to see this for yourself, you could change `{ "remove": { "field": [ "agent" ], "ignore_missing": true, "ignore_failure": true } },` to instead rename `agent` to something like `agentorig` and then compare `agentorig` to the Wazuh `agent` data to see that it is totally different.

**Reply by lmdeaton1** [reply]
OK, I understand now.  The "agent" field is different from the "message.agent" field.  Thanks for the clarification.
